drivers/include/w5500.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2023 Stefan Schmidt
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_w5500 W5500 ethernet driver
+ * @ingroup     drivers_netdev
+ * @brief       Driver for W5500 ethernet devices
+ *
+ * This device driver only exposes the MACRAW mode of W5500 devices, so it does
+ * not offer any support for the on-chip IPv4, UDP, and TCP capabilities of
+ * these chips. In connection with RIOT we are only interested in the RAW
+ * Ethernet packets, which we can use through netdev with any software network
+ * stack provided by RIOT (e.g. GNRC). This enables W5500 devices to communicate
+ * via IPv6, enables unlimited connections, and more...
+ *
+ * @note        This driver can be used in polling or interrupt mode.
+ *              On some shields the interrupt line is not enabled by default,
+ *              you have to close the corresponding solder bridge to make the
+ *              interrupt mode work...
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the W5500 device driver
+ *
+ * @author      Stefan Schmidt <stemschmidt@gmail.com>
+ */
+
+#ifndef W5500_H
+#define W5500_H
+
+#include <stdint.h>
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+#include "net/netdev.h"
+#include "kernel_defines.h"
+#include "ztimer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   W5500 device descriptor
+ */
+typedef struct {
+    spi_t spi;                    /**< SPI bus used */
+    spi_clk_t clk;                /**< clock speed used on the selected SPI bus */
+    gpio_t cs;                    /**< pin connected to the chip select line */
+    gpio_t evt;                   /**< pin connected to the INT line */
+    uint32_t polling_interval_ms; /**< interval for polling, 0 if interrupt mode */
```

Comment:
Have you considered doing a compile-time check, so we don't even need to compile the polling code in if we have a IRQ pin connected?

```suggestion
#if W5500_PARAM_EVT == GPIO_UNDEF
    uint32_t polling_interval_ms; /**< interval for polling, 0 if interrupt mode */
#endif
```

Reply:
The w5500.h header is independent of the configuration, it does not include w5500_params.h. So I do not have the information W5500_PARAM_IRQ == GPIO_UNDEF available in the header